components/post_view/post_list.jsx

```diff
@@ -347,43 +333,45 @@ export default class PostList extends React.PureComponent {
     }
 
     checkBottom = () => {
-        if (!this.refs.postlist) {
+        if (!this.postListRef.current) {
             return true;
         }
+        const postlistScrollHeight = this.postListRef.current.scrollHeight;
+        const postlistClientHeight = this.postListRef.current.clientHeight;
 
         // No scroll bar so we're at the bottom
-        if (this.refs.postlist.scrollHeight <= this.refs.postlist.clientHeight) {
+        if (postlistScrollHeight <= postlistClientHeight) {
             return true;
         }
 
-        return this.refs.postlist.clientHeight + this.refs.postlist.scrollTop >= this.refs.postlist.scrollHeight - CLOSE_TO_BOTTOM_SCROLL_MARGIN;
+        return postlistClientHeight + this.postListRef.current.scrollTop >= postlistScrollHeight - CLOSE_TO_BOTTOM_SCROLL_MARGIN;
     }
 
     handleWindowResize = () => {
         this.handleResize();
     }
 
     handleResize = (forceScrollToBottom) => {
-        const postList = this.refs.postlist;
-        const messageSeparator = this.refs.newMessageSeparator;
-        const doScrollToBottom = this.atBottom || forceScrollToBottom;
-
-        if (postList) {
-            if (doScrollToBottom) {
-                postList.scrollTop = postList.scrollHeight;
-            } else if (!this.hasScrolled && messageSeparator) {
-                const element = ReactDOM.findDOMNode(messageSeparator);
-                element.scrollIntoView();
-            }
+        if (this.resizeAnimationFrame) {
+            window.cancelAnimationFrame(this.resizeAnimationFrame);
```

Comment:
Also, we're not clearing the value of `this.resizeAnimationFrame` when the callback is fired so it will always try to cancel the previous callback even if it's already been called

Reply:
Though the function says resize it was being used to call for any postList scroll change actions(Just the way it was). 

We don't need to clear the value for cancelling the animationFrame. It is similar to timeout the value exists if there is a pending animation frame.

Reply:
Since it's in an if statement, I would expect it to only call `cancelAnimationFrame` some of the time, but because it doesn't clear the value of `this.resizeAnimationFrame`, it will "cancel" it basically every time. It works fine and there's nothing majorly wrong with it, but the code is written in a way that's a bit confusing